### PR TITLE
PHOENIX-6026: Fix BackwardCompatibilityIT so it can run locally

### PR DESCRIPTION
**Pull request by ChinmaySKulkarni**

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/BackwardCompatibilityIT.java [hunk 1]
```diff
@@ -118,6 +125,7 @@ public synchronized void doSetup() throws Exception {
         zkQuorum = "localhost:" + hbaseTestUtil.getZkCluster().getClientPort();
         url = PhoenixRuntime.JDBC_PROTOCOL + PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + zkQuorum;
         DriverManager.registerDriver(PhoenixDriver.INSTANCE);
+        checkForPreConditions();
```
Comment (sguggilam) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/BackwardCompatibilityIT.java:128]: We need to have this before each test as it checks for the non existence of any SYSTEM tables before each test run to make sure that there is no existing metadata

**Comment by sguggilam** [comment]
Just one comment, else looks good to me

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/BackwardCompatibilityIT.java (hunk 1 above)
Comment (ChinmaySKulkarni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/BackwardCompatibilityIT.java:128]: Yes that's what is happening. This is now called in `doSetup()` which will be invoked before each test i.e. it is marked as `@Before`

Comment (sguggilam) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/BackwardCompatibilityIT.java:128]: Makes sense, thanks

**Pull request closed without merging**